src/views/Scenario/Scenario.js

```diff
@@ -102,6 +104,45 @@ const Scenario = (props) => {
     : t('views.scenario.dropdown.scenario.label', Scenario);
   const showBackdrop = currentScenario.status === STATUSES.LOADING;
 
+  // App Insigths
+  const appInsights =
+    ENABLE_APPLICATION_INSIGHTS
+      ? AppInsights.getInstance()
+      : undefined;
+
+  // Track create scenario
+  const trackCreateScenario = () => {
+    appInsights.trackEvent({ name: 'CreateScenario' }, { name: currentScenario.rootId });
+    appInsights.trackMetric({ name: 'CreateScenarioValue', average: 1, sampleCount: 1 });
+  };
+
+  const createTrackedScenario = (workspaceId, scenario) => {
+    if (appInsights) {
+      trackCreateScenario();
+    }
+    createScenario(workspaceId, scenario);
+  };
+
+  // Track launch scenario
+  const trackLaunchScenario = () => {
+    appInsights.trackEvent({ name: 'LaunchScenario' }, { scenarioId: currentScenario.rootId });
+    appInsights.trackMetric({ name: 'LaunchScenarioValue', average: 1, sampleCount: 1 });
+  };
+
+  const updateAndLaunchTrackedScenario = (workspaceId, scenarioId, scenarioParameters) => {
+    if (appInsights) {
+      trackLaunchScenario();
+    }
+    updateAndLaunchScenario(workspaceId, scenarioId, scenarioParameters);
+  };
+
+  const launchTrackedScenario = (workspaceId, scenarioId) => {
+    if (appInsights) {
+      trackLaunchScenario();
+    }
+    launchScenario(workspaceId, scenarioId);
+  };
```

Comment:
Could we add the tracking part directly in redux sage instead of wrapping the calls to `createScenario`, `updateAndLaunchScenario` and `launchScenario` here ?